From 56cdacb669a182e69159b840a6a440f7f19211f9 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 4 Jul 2016 20:14:52 -0700
Subject: [PATCH 4/4] clang: Check for float ABI on arm

Triggering hard-float on arm does not just
depend upong triplet being *-*-eabihf but
also check for float ABI which can be
set on commandline

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 lib/Driver/ToolChains.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index bcd91f0..fd31d7e 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -4212,7 +4212,9 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     default:
       ArchName = Triple.getArchName().str();
     }
-    if (Triple.getEnvironment() == llvm::Triple::MuslEABIHF)
+    if (Triple.getEnvironment() == llvm::Triple::MuslEABIHF ||
+        (Triple.getEnvironment() == llvm::Triple::MuslEABI &&
+        tools::arm::getARMFloatABI(*this, Args) == tools::arm::FloatABI::Hard))
       ArchName += "hf";
 
     return "/lib/ld-musl-" + ArchName + ".so.1";
-- 
2.9.0

